### PR TITLE
chore: upgrade to celestia-app v0.7.0

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-node/issues/1142
Opens https://github.com/celestiaorg/celestia-node/issues/1231

### CODE REVIEW

File: header/pb/extended_header.proto
Comment (rootulp) [inline header/pb/extended_header.proto:13]: [for reviewers] this type moved from celestia-core to celestia-app in https://github.com/celestiaorg/celestia-app/pull/706

File: header/store/heightsub.go
Comment (rootulp) [inline header/store/heightsub.go:82]: [for reviewers] I encountered an error after running `make fmt` where `log` was declared twice in this package because [this line](https://github.com/rootulp/celestia-node/blob/100bfe33231a4574878a4b0b4244f0adf978aa08/header/store/store.go#L19) so I added an import alias here.

File: ipld/plugin/nmt.go
Comment (rootulp) [inline ipld/plugin/nmt.go:34]: [for reviewers] all the `consts` from celestia-core moved to `appconsts` in celestia-app

**Comment by rootulp** [comment]
Integ test failed CI. Will debug tomorrow

**Comment by Wondertan** [comment]
@rootulp, is it ready for review?

**Comment by rootulp** [comment]
@Wondertan nope, sorry I haven't investigated why integ test is failing yet.

**Comment by Wondertan** [comment]
Note that we have one flakey integration test that may be annoying for you to debug

**Comment by rootulp** [comment]
Noted. I just started a third retry

**Comment by rootulp** [comment]
Notes from investigation:
- This log is suspect: `starting Bridge Node: square size is not a power of two: 0` ([link](https://github.com/celestiaorg/celestia-node/actions/runs/3102238511/jobs/5033883608#step:4:4583)). The error appears to originate from celestia-app [`shares.Split`](https://github.com/celestiaorg/celestia-app/blob/d53f66b67946d20af5a24487c2f79da4ebb1b171/pkg/shares/share_splitting.go#L26) even though the block size in this test is `16` per [this line](https://github.com/rootulp/celestia-node/blob/100bfe33231a4574878a4b0b4244f0adf978aa08/node/tests/reconstruct_test.go#L40).

**Comment by Wondertan** [comment]
Thanks, @rootulp. Mockable App is something we have wanted for a while already, and it just has never been a priority. From your message, it seems like it is blocking the update now, right?

**Comment by rootulp** [comment]
Yep blocking. @evan-forbes is working on a slightly different approach to unblock this PR.

File: header/store/heightsub.go
Comment (rootulp) [reply header/store/heightsub.go]: Resolved in https://github.com/celestiaorg/celestia-node/pull/1147/commits/90c9d2f53276072abec834f80c36f8f62901bc24

File: service/rpc/share.go
Comment (rootulp) [reply service/rpc/share.go]: Resolved in https://github.com/celestiaorg/celestia-node/pull/1147/commits/90c9d2f53276072abec834f80c36f8f62901bc24

**Comment by rootulp** [comment]
Blocked on https://github.com/celestiaorg/celestia-app/issues/770

**Comment by rootulp** [comment]
Update: blocked on https://github.com/celestiaorg/celestia-node/issues/628 and https://github.com/celestiaorg/celestia-node/pull/1160

**Comment by Wondertan** [comment]
@rootulp, the https://github.com/celestiaorg/celestia-node/pull/1160 is now merged, so you freely continue here.

Ping if anything is needed from our side

File: go.mod
Comment (rootulp) [inline go.mod:10]: https://github.com/celestiaorg/celestia-app/releases/tag/v0.7.0-rc3 doesn't contain the new testnode used for swamp tests so we either want to:
1. patch rc3 with it
2. upgrade to [v0.7.0-rc5](https://github.com/celestiaorg/celestia-app/releases/tag/v0.7.0-rc5)

will opt for option 2

Comment (rootulp) [reply go.mod]: Option 2 won't work as-is because there are changes on branch evan/v0.7.0-rc-1-test-node (e.g. [FillBlock](https://github.com/celestiaorg/celestia-app/blob/evan/v0.7.0-rc-1-test-node/testutil/testnode/interaction_api.go#L83)) that we need to patch onto v0.7.0-rc5 or merge onto `main` and cut a new RC

@evan-forbes

Comment (rootulp) [reply go.mod]: created https://github.com/celestiaorg/celestia-app/issues/860

**Comment by rootulp** [comment]
This green check is a bit misleading. I expected unit tests passed but it looks like they didn't run https://github.com/celestiaorg/celestia-node/pull/1147/checks?sha=fdf74534240abc0b40824bf7bf8f974b15f55fcb

## Screenshot

Screen Shot 2022-10-12 at 11 03 54 PM

File: header/header.go
Comment (rootulp) [inline header/header.go:52]: In contradiction to https://github.com/celestiaorg/celestia-app/blob/dd02bb8919ec7c093784c49795c956ddb8cd2edf/pkg/shares/share_splitting.go#L24

I think this may need to be false to preserve backwards compatibility with blocks that have malleated transactions without share indexes.

```suggestion
		shares, err := appshares.Split(b.Data, false)
```

Tagging @evan-forbes for confirmation

File: go.mod
Comment (rootulp) [inline go.mod:11]: [for reviewers] we should probably merge https://github.com/celestiaorg/celestia-app/pull/866 cut v0.7.0-rc6 and upgrade to that here before merging this PR

File: header/store/heightsub.go
Comment (Wondertan) [reply header/store/heightsub.go:82]: Weird. It wouldn't compile before then

File: nodebuilder/tests/swamp/swamp_tx.go
Comment (renaynay) [inline nodebuilder/tests/swamp/swamp_tx.go:15]: Shouldn't this method still somehow continue to respect `blocks` param? 
```
produces the given amount of contiguous blocks
```

File: share/availability/light/availability_test.go
Comment (renaynay) [inline share/availability/light/availability_test.go:302]:
```suggestion
```

File: share/availability/test/testing.go
Comment (renaynay) [inline share/availability/test/testing.go:21]:
```suggestion
	"github.com/celestiaorg/celestia-app/pkg/da"
	
```

File: share/availability/light/availability_test.go
Comment (renaynay) [inline share/availability/light/availability_test.go:301]: I'm fine making an issue for this but we should do it sooner rather than later to keep tests up to date.

File: share/availability/light/availability_test.go
Comment (renaynay) [inline share/availability/light/availability_test.go:289]: I've never seen this param before @evan-forbes can you explain how it works? Also, tests work if we decrease OG square size to 8.

File: header/header.go
Comment (renaynay) [reply header/header.go:52]: So we're deciding *not* to preserve block history on arabica, right?

File: share/availability/light/availability_test.go
Comment (Wondertan) [reply share/availability/light/availability_test.go:301]: We can put true there already btw. There is no transaction logic involved in the test

File: nodebuilder/tests/swamp/swamp.go
Comment (Wondertan) [reply nodebuilder/tests/swamp/swamp.go:82]: We should because we have to control block time. https://github.com/celestiaorg/celestia-node/pull/1147#discussion_r994431647

File: header/header.go
Comment (Wondertan) [reply header/header.go:52]: Yes, from our sync discussion

File: nodebuilder/tests/swamp/swamp.go
Comment (evan-forbes) [reply nodebuilder/tests/swamp/swamp.go]: :+1: 
https://github.com/celestiaorg/celestia-node/pull/1147/commits/4f789c8dbd9fc751aca636de29290ec07d592010

File: nodebuilder/tests/swamp/swamp.go
Comment (evan-forbes) [reply nodebuilder/tests/swamp/swamp.go]: :+1: https://github.com/celestiaorg/celestia-node/pull/1147/commits/41cb3f9116b39567c0a312b73d9447127996fed2

File: nodebuilder/tests/swamp/swamp_tx.go
Comment (evan-forbes) [reply nodebuilder/tests/swamp/swamp_tx.go]: oh shiieet, whoops good catch!! https://github.com/celestiaorg/celestia-node/pull/1147/commits/ce0f8341a8be2293f0bb6258973116d29f398798

File: share/availability/light/availability_test.go
Comment (evan-forbes) [reply share/availability/light/availability_test.go:289]: This square size is effectively a cached value that is normally filled out by block producers, so unless we want to have an instance of the application producer the block for us, we have to fill it out by hand.

Comment (evan-forbes) [reply share/availability/light/availability_test.go:289]: come to think of it, a lot of the logic is stateless, so we should be able to abstract it out for anyone to call... :thinking:

File: share/availability/light/availability_test.go
Comment (Wondertan) [reply share/availability/light/availability_test.go:301]: Well, the logic that we are testing here does not have anything to do with the content of the shares. It just passes shares back and forth and disconnected from the App's share verification logic so I would put true here already.

File: header/store/heightsub.go
Comment (rootulp) [reply header/store/heightsub.go]: On second pass I see what happened: `make fmt` is automatically adding an import for `"log"` but it shouldn't be imported and instead [this line](https://github.com/rootulp/celestia-node/blob/100bfe33231a4574878a4b0b4244f0adf978aa08/header/store/store.go#L19) should be used.

Comment (rootulp) [reply header/store/heightsub.go]: Resolved in https://github.com/celestiaorg/celestia-node/pull/1147/commits/e321b855f99df64941e6124fa9b88fa2884e2e63

**Comment by rootulp** [comment]
Another instance of https://github.com/celestiaorg/celestia-node/issues/1090 encountered [here](https://github.com/celestiaorg/celestia-node/actions/runs/3243493927/jobs/5318289227#step:4:279)

File: share/availability/test/testing.go
Comment (rootulp) [reply share/availability/test/testing.go:21]: Ou of curiosity, what motivates this suggestion? Does celestia-node prefer to separate some imports with newlines? If so, which imports?

Related: it looks like https://golangci-lint.run/usage/linters/#goimports doesn't have a config option for newlines between particular import statements but it may speed up reviews if maintainers could rely on a linter for this

File: share/availability/light/availability_test.go
Comment (rootulp) [reply share/availability/light/availability_test.go:301]: Putting `true` here will fail these unit tests because [this error](https://github.com/celestiaorg/celestia-app/blob/906fdf74f871f13d3fe0c632dfda95db494521ee/pkg/shares/share_splitting.go#L130)

Comment (Wondertan) [reply share/availability/light/availability_test.go:301]: Ok, so we cannot even split shares, unless we generate them with some special logic.

File: share/availability/test/testing.go
Comment (rootulp) [reply share/availability/test/testing.go:21]: Reverted this change b/c it fails lint https://github.com/celestiaorg/celestia-node/pull/1147/commits/44c8d92cc672c9cd04dc5d939a00687f55d4bc32

File: header/header.go
Comment (rootulp) [reply header/header.go:52]: Sorry I missed the sync discussion but based on comments, I think this is resolvable and we will not preserve backwards compatibility on Arabica.

File: share/availability/light/availability_test.go
Comment (rootulp) [reply share/availability/light/availability_test.go:301]: > I'm fine making an issue for this

I opened https://github.com/celestiaorg/celestia-node/issues/1231 if reviewers are comfortable merging this PR as-is and following-up with that issue

File: nodebuilder/tests/swamp/swamp.go
Comment (evan-forbes) [reply nodebuilder/tests/swamp/swamp.go]: https://github.com/celestiaorg/celestia-node/pull/1147/commits/4f789c8dbd9fc751aca636de29290ec07d592010

File: share/availability/light/availability_test.go
Comment (renaynay) [reply share/availability/light/availability_test.go:301]: I'm fine with that.

Comment (Wondertan) [reply share/availability/light/availability_test.go:301]: +

File: nodebuilder/tests/swamp/config.go
Comment (Wondertan) [inline nodebuilder/tests/swamp/config.go:21]: optional nit: It's time to rename this to just Config, as there are no more dependencies/components. I minor thing to do that I can create an issue for

File: nodebuilder/tests/swamp/swamp.go
Comment (Wondertan) [inline nodebuilder/tests/swamp/swamp.go:75]: We can also remove this TODO. It's fine to make tested App assumption here and there is nothing TBD about this

**Review by Wondertan — APPROVED** [review]
LGTM. Thank you for the help ❤️ 

Left two non-blocking optional nits

File: share/availability/light/availability_test.go
Comment (renaynay) [reply share/availability/light/availability_test.go:289]: Okay I see why OriginalSquareSize also works with 8 -- bc with 16, it had more tail share padding.

Comment (renaynay) [reply share/availability/light/availability_test.go:289]: feel free to resolve @evan-forbes

**Review by renaynay — APPROVED** [review]

**Pull request merged**
